### PR TITLE
chore(ai): adding UA string for AIConversation

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adds custom UA string for AIConversation component. Bumps the minimum aws-amplify version to 6.6.5 for all packages. 

#### Description of how you validated changes

CleanShot 2024-10-16 at 20 47 37@2x

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jordanvn — APPROVED** [review]
MSTM, once test failures are resolved

File: packages/react-ai/src/types.ts [hunk 1]
```diff
@@ -14,20 +15,28 @@ export type TextContent = NonNullable<ConversationMessageContent['text']>;
 
 export type ImageContent = NonNullable<ConversationMessageContent['image']>;
 
-export type InputContent = Parameters<
-  Conversation['sendMessage']
->[0]['content'][number];
+// Note: the conversation sendMessage function is an overload
+// that accepts a stirng OR an object
```
Comment (jordanvn) [inline packages/react-ai/src/types.ts:19]: Nit:
```suggestion
// that accepts a string OR an object
```

File: packages/react-ai/src/version.ts
Comment (ashika112) [inline packages/react-ai/src/version.ts:1]: is this something new for UI repo?

**Review by ashika112 — APPROVED** [review]

File: packages/react-ai/src/version.ts
Comment (dbanksdesign) [reply packages/react-ai/src/version.ts:1]: Not anything new, a pattern we have for all UI packages where we use the UA string. We have a script to update the version number on build. This is so we don't have to import package.json

**Pull request merged**